docs/adr/002.md

```diff
@@ -0,0 +1,261 @@
+  custom dynamics. Or at least, {mod}`tensorwaves` does not understand those
+  custom dynamics.
+- [ADR-001](./001.md): parameters _and_ variables are to be expressed as
+  `sympy.Symbol`s.
+- [#454](https://github.com/ComPWA/expertsystem/pull/454): dynamics are
+  specified as a mapping of `sympy.Function` to a `sympy.Expr`, but now there
+  is no way to supply those `sympy.Expr`s with expected `sympy.Symbol`s
+  (parameters and variables).
+
+### Issues with existing set-up
+
+- There is no clear way to apply dynamics functions to a specific decaying
+  particle, that is, to a specific edge of the `StateTransitionGraph`s (`STG`).
+  Currently, we just work with a mapping of `Particle`s to some dynamics
+  expression, but this is not feasible when there there are identical particles
+  on the edges.
+- The set of variables to which a dynamics expression applies, is determined by
+  the position within the `STG` that it is applied to. For instance, a
+  relativistic Breit-Wigner that is applied to the resonance in some 1-to-3
+  isobar decay (described by an `STG` with final state edges 2, 3, 4 and
+  intermediate edge 1) would work on the invariant mass of edge 3 and 4
+  (`mSq_3+4`).
+- Just like variables, parameters need to be identifiable from their position
+  within the `STG` (take a relativistic Breit-Wigner _with form factors_, which
+  would require break-up momentum as a parameter), but also require some
+  suggested starting value (e.g. expected pole position). These starting values
+  are usually taken from the edge and node properties within the `STG`.
+
+## Decision drivers
+
+The following points are nice to have or can influence the decision but are not
+essential and can be part of the users responsibility.
+
+1. The parameters that a dynamics functions requires, are registered
+   automatically and linked together.
+2. Kinematic variables used in dynamics functions are also linked
+   appropriately.
+3. It is easy to define custom dynamics (no boilerplate code).
+
+### Solution requirements
+
+1. It is easy to apply dynamics to specific components of the `STG`s. Note:
+   it's important that the dynamics can be applied to resonances of some
+   **selected** graphs and not generally all graphs in which the resonance
+   appears.
+2. Where possible, suggested (initial) parameter values are provided as well.
+3. It is possible to use and inspect the dynamics expression itself
+   independently from the `expertsystem`.
+4. Follow open-closed principle. Probably the most important decision driver.
+   The solution should be flexible enough to handle any possible scenario,
+   without having to change the interface defined in requirement 1!
+
+## Considered solutions
+
+To satisfy [**requirement 1**](#solution-requirements), we propose the
+following syntax:
+
+```python
+# model: ModelInfo
+# graph: StateTransitionGraph
+model.set_dynamics(graph, edge_id=1, expression_builder)
```

Comment:
This `expression_builder` is still a part, I'm quite hesitant about, as this puts a lot of responsibility on the design of how to handle the connection of kinematic variables and parameters. I would simply turn this into an expression, and let some independent code be responsible for the correct construction of that expression (correct meaning creating the correct symbols so that they are connected to other ones in the full model).

Reply:
see https://github.com/ComPWA/expertsystem/pull/461/commits/1241b5cfdbcb68692a34bb5c440ff6a6029fe4cc